core/src/main/kotlin/com/strumenta/kolasu/codegen/ASTCodeGenerator.kt

```diff
@@ -30,12 +30,15 @@ abstract class ASTCodeGenerator<R : Node> {
     }
 
     fun printToString(ast: R): String {
-        return PrinterOutput(this.nodePrinters, nodePrinterOverrider)
-            .apply { this.print(ast) }
-            .text()
+        return printerOutput().apply { this.print(ast, prefix, postfix) }.text()
     }
 
     fun printToFile(root: R, file: File) {
         file.writeText(printToString(root))
     }
+
+    open fun printerOutput() = PrinterOutput(this.nodePrinters, nodePrinterOverrider)
+
+    open val prefix: String get() = ""
```

Comment:
Why would someone need prefix and postfix? Could you indicate some examples?

Reply:
Prefix and postfix are exposed as parameters by the printer output, but `ASTCodeGenerator`doesn't allow to pass them. Anyway for prefix I have a couple of use cases such as adding information at the top of the file (e.g. licensing, or notes about the file having been generated, etc.) and adding for example a shebang line for scripts (or for a client the `#! CLIENT_NAME_REDACTED-java` header)

Reply:
Oh, right, it makes sense. Can you add this as a comment?